src/main/java/Duke.java

```diff
@@ -1,10 +1,241 @@
+        }
+    }
+
+    /**
+     * Return void.
+     * Function is responsible for printing out the whole task list of the user.
+     *
+     * @param tasks list of tasks input by user
+     */
+    public static void printList(Task[] tasks) {
+        int count = 0;
+        System.out.println("Here is your list:");
+        for (Task item : tasks) {
+            if (tasks[count] != null) {
+                count++;
+                System.out.println(count + ". " + item);
+            }
+        }
+    }
+
+    public static boolean hasListKeyword(String arg) {
+        return arg.trim().matches("^list$");
+    }
+
+    public static boolean hasDoneKeyword(String arg) {
+        return arg.trim().matches("^[done]+\\s+[0-9]+$");
+    }
+
+    public static boolean hasTodoKeyword(String arg) {
+        return arg.trim().toLowerCase().contains("todo");
+    }
+
+    public static boolean hasDeadlineKeyword(String arg) {
+        return arg.trim().toLowerCase().contains("deadline");
+    }
+
+    public static boolean hasEventKeyword(String arg) {
+        return arg.trim().toLowerCase().contains("event");
+    }
+
+    /**
+     * Returns the required value for keyword which is the first word keyed in by user.
+     *
+     * @param query user raw data input.
+     * @return keyword value from Keyword enum class.
+     */
+    public static Keyword getKeywordStatus(String query) {
+        Keyword keyword;
+        if (hasDoneKeyword(query)) {
+            keyword = Keyword.DONE_TASK;
+        } else if (hasTodoKeyword(query)) {
+            keyword = Keyword.TODO_TASK;
+        } else if (hasDeadlineKeyword(query)) {
+            keyword = Keyword.DEADLINE_TASK;
+        } else if (hasEventKeyword(query)) {
+            keyword = Keyword.EVENT_TASK;
+        } else if (hasListKeyword(query)) {
+            keyword = Keyword.LIST_ITEMS;
+        } else if (query.trim().equals("bye")) {
+            keyword = Keyword.GOODBYE_KEYWORD;
+        } else {
+            keyword = Keyword.NO_KEYWORD;
+        }
+        return keyword;
+    }
+
+    public static void waitForQuery() {
+        String query = "";
+        Scanner userInput = new Scanner(System.in);
+
+        while (!query.equals("bye")) {
+            System.out.print("=>");
+            if (userInput.hasNextLine()) {
+                query = userInput.nextLine();
+            }
+            addTask(query);
+        }
+    }
+
+    /**
+     * Returns void.
+     * Function is responsible for adding different Tasks to the task list.
+     *
+     * @param query user raw data input
+     * @throws NullPointerException if user keys in done [number] when there is no such task.
+     */
+    public static void addTask(String query) throws NullPointerException {
+        Keyword keyword = getKeywordStatus(query);
+        switch (keyword) {
+        case DONE_TASK:
+            try {
+                int taskNumber = getTaskNum(query);
+                tasks[taskNumber - 1].markAsDone();
```

Comment:
Perhaps taskNumber - 1 can be assigned to a variable to inform users that you are searching by index instead? (maybe call it taskIndex or sth)